thamos/config.py

```diff
@@ -49,6 +50,54 @@
 
 # The schema is enforcing all the options. This will make sure the right version of Thamos is
 # installed and no configuration options are silently ignored.
+_CONFIG_RUNTIME_ENVIRONMENT_SCHEMA = {
+    "type": "object",
+    "properties": {
+        "cuda_version": {"type": ["string", "null"]},
+        "openblas_version": {"type": ["string", "null"]},
+        "openmpi_version": {"type": ["string", "null"]},
+        "cudnn_version": {"type": ["string", "null"]},
+        "mkl_version": {"type": ["string", "null"]},
+        "base_image": {"type": ["string", "null"]},
+        "hardware": {
+            "type": "object",
+            "properties": {
+                "cpu_family": {"type": "integer"},
+                "cpu_model": {"type": "integer"},
+                "gpu_model": {"type": ["string", "null"]},
+            },
+            "required": [],
+            "additionalProperties": False,
+        },
+        "name": {"type": "string", "pattern": r"^[a-zA-Z0-9:_-]+$"},
+        "operating_system": {
+            "type": "object",
+            "properties": {
+                "name": {"type": ["string", "null"]},
+                "version": {"type": ["string", "null"]},
+            },
+            "required": ["name", "version"],
+            "additionalProperties": False,
+        },
+        "platform": {"type": ["string", "null"]},
+        "python_version": {
+            "type": "string",
+            "pattern": r"^[0-9]+\.[0-9]+$",
+        },
+        "recommendation_type": {
+            "type": "string",
+            "enum": [
+                "latest",
+                "stable",
+                "performance",
+                "security",
+                "testing",
+            ],
+        },
+    },
+    "additionalProperties": False,
+    "required": ["name"],
```

Comment:
Only 'name'? for setting a new runtime environment don't we want to add other parameters required? What happens if for one runtime environment we have only the name and there is an overlay with the same name?

Reply:
> Only 'name'? for setting a new runtime environment don't we want to add other parameters required? 

Yes, if no other parameters are supplied, we do "generic" resolution for any runtime environment.

> What happens if for one runtime environment we have only the name and there is an overlay with the same name?

That is a configuration issue. `thamos check` should fail.

Reply:
> > What happens if for one runtime environment we have only the name and there is an overlay with the same name?
> 
> That is a configuration issue. `thamos check` should fail.

If you want to manage Pipenv files, currently, we do not handle Pipfile content at all. Might be a good idea to add such handling logic if we plan to substitute Pipenv completely though.

Reply:
Do you want to open new issue? :)

Reply:
Yes, why not. :+1:

Reply:
> Yes, why not.

Thanks 🍻